src/main/java/com/example/Restaurant_Management_System_REST_API/service/MenuRecordService.java

```diff
@@ -0,0 +1,77 @@
+package com.example.Restaurant_Management_System_REST_API.service;
+
+import com.example.Restaurant_Management_System_REST_API.DTO.MenuRecordDTORequest;
+import com.example.Restaurant_Management_System_REST_API.DTO.MenuRecordDTOResponse;
+import com.example.Restaurant_Management_System_REST_API.exception.NotFoundInDatabaseException;
+import com.example.Restaurant_Management_System_REST_API.model.entity.MenuRecord;
+import com.example.Restaurant_Management_System_REST_API.repository.MenuRecordRepository;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class MenuRecordService implements GenericBasicCrudOperations<MenuRecordDTOResponse, MenuRecordDTORequest, Long> {
+
+    private MenuRecordRepository menuRecordRepository;
+    private ModelMapper modelMapper;
+
+    @Override
+    public MenuRecordDTOResponse create(MenuRecordDTORequest menuRecordDTORequest) {
+        MenuRecord menuRecord = modelMapper.map(menuRecordDTORequest, MenuRecord.class);
+        menuRecordRepository.save(menuRecord);
+
+        return modelMapper.map(menuRecord, MenuRecordDTOResponse.class);
+    }
+
+    @Override
+    public MenuRecordDTOResponse findById(Long menuRecordId) throws NotFoundInDatabaseException {
+        MenuRecord menuRecord = menuRecordRepository.findById(menuRecordId).orElseThrow(() ->
+                new NotFoundInDatabaseException(MenuRecord.class));
+
+        return modelMapper.map(menuRecord, MenuRecordDTOResponse.class);
+    }
+
+    @Override
+    public List<MenuRecordDTOResponse> findAll() {
+        Iterable<MenuRecord> iterable = menuRecordRepository.findAll();
+        List<MenuRecordDTOResponse> list = new ArrayList<>();
+
+        for (MenuRecord menuRecord : iterable) {
+            list.add(modelMapper.map(menuRecord, MenuRecordDTOResponse.class));
+        }
+//        iterable.forEach(list::add); a shorter version
```

Comment:
but it turns out I cannot make it like that because I was trying to add MenuRecord objects to a List<MenuRecordDTOResponse>. So I stay with it as it was

Reply:
Because you were missing mapping. Do you want to leave it like that? Feel free to do as you like.

Reply:
yes I know I'm missing mapping, but I could do it only with stream but the result is not what I want -> it takes more place than original code.

I wonder if I can map somehow using method reference, but it seems that I'm not able to do it.

is there a way to add mapping using method reference and this line of code?
iterable.forEach(list::add)

Reply:
okay I think I could do something like that?

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/824c6e47-e9e5-4cfd-8401-da4bbd7984b2)

what do you think?

Reply:
Yep, this one's looking good!

Reply:
ps. It works good so I think it's great!